src/base/bittorrent/peerinfo.cpp

```diff
@@ -181,6 +181,28 @@ QString PeerInfo::client() const
     return QString::fromStdString(m_nativeInfo.client);
 }
 
+QString PeerInfo::peerId() const
+{
+    // when peer ID is not known yet it contains only zero bytes,
+    // do not create string in such case, return empty string instead
+    if (m_nativeInfo.pid.is_all_zeros())
+        return {};
+
+    // interesting part of a typical peer ID is first 8 chars
+    static constexpr int idCharCount = std::min(8, static_cast<int>(m_nativeInfo.pid.size()));
+
+    // check that peer ID slice consists only of printable ASCII characters,
+    // this should filter out most of the improper IDs
+    for (int i = 0; i < idCharCount; ++i)
+    {
+        unsigned char const c = m_nativeInfo.pid[i];
+        if (c < 32 || c > 126)
+            return tr("Unknown");
+    }
+
+    return QString::fromLatin1(m_nativeInfo.pid.data(), idCharCount);
```

Comment:
There are numerous coding style violation.
Also you should make good use of the loop. Please apply this:
```suggestion
    QString id;
    for (int i = 0; i < idCharCount; ++i)
    {
        const unsigned char c = m_nativeInfo.pid[i];
        if ((c < 32) || (c > 126))
            return tr("Unknown");
        id += QChar::fromLatin1(c);
    }

    return id;
```